Source/scrollrt.cpp

```diff
@@ -1552,42 +1518,40 @@ void TilesInView(int *rcolumns, int *rrows)
 
 void CalcViewportGeometry()
 {
-	tileShiftX = 0;
-	tileShiftY = 0;
+	tileShift = { 0, 0 };
 
 	// Adjust by player offset and tile grid alignment
 	int xo = 0;
 	int yo = 0;
 	CalcTileOffset(&xo, &yo);
-	tileOffsetX = 0 - xo;
-	tileOffsetY = 0 - yo - 1 + TILE_HEIGHT / 2;
+	tileOffset = { -xo, -yo - 1 + TILE_HEIGHT / 2 };
 
 	TilesInView(&tileColums, &tileRows);
 	int lrow = tileRows - RowsCoveredByPanel();
 
 	// Center player tile on screen
-	ShiftGrid(&tileShiftX, &tileShiftY, -tileColums / 2, -lrow / 2);
+	ShiftGrid(&tileShift.deltaX, &tileShift.deltaY, -tileColums / 2, -lrow / 2);
 
 	tileRows *= 2;
 
 	// Align grid
 	if ((tileColums & 1) == 0) {
-		tileShiftY--; // Shift player row to one that can be centered with out pixel offset
+		tileShift.deltaY--; // Shift player row to one that can be centered with out pixel offset
 		if ((lrow & 1) == 0) {
 			// Offset tile to vertically align the player when both rows and colums are even
 			tileRows++;
-			tileOffsetY -= TILE_HEIGHT / 2;
+			tileOffset.deltaY -= TILE_HEIGHT / 2;
 		}
 	} else if ((tileColums & 1) != 0 && (lrow & 1) != 0) {
 		// Offset tile to vertically align the player when both rows and colums are odd
-		ShiftGrid(&tileShiftX, &tileShiftY, 0, -1);
+		ShiftGrid(&tileShift.deltaX, &tileShift.deltaY, 0, -1);

```

Comment:
```suggestion
		tileShift += DIR_W;
```

Reply:
Double-check this one. I think it's `DIR_N`.

Reply:
Good catch